src/python_testing/TC_CADMIN_1_5.py

```diff
@@ -0,0 +1,212 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import asyncio.exceptions as ae
+from time import sleep
+
+import chip.clusters as Clusters
+import chip.interaction_model
+from chip import ChipDeviceCtrl
+from chip.exceptions import ChipStackError
+from chip.native import PyChipError
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mdns_discovery import mdns_discovery
+from mobly import asserts
+
+
+class TC_CADMIN_1_5(MatterBaseTest):
+    async def get_txt_record(self):
+        discovery = mdns_discovery.MdnsDiscovery(verbose_logging=True)
+        comm_service = await discovery.get_commissionable_service(
+            discovery_timeout_sec=240,
+            log_output=False,
+        )
+        return comm_service
+
+    async def CommissionOnNetwork(self, setup_code: int, discriminator: int):
+        ctx = asserts.assert_raises(ChipStackError)
+        try:
```

Comment:
This logic is a bit weird - could you just remove this and have the expected error be passed in? Or just use two different functions?